src/main/kotlin/net/minecrell/pluginyml/PlatformPlugin.kt

```diff
@@ -56,10 +58,8 @@ abstract class PlatformPlugin<T : PluginDescription>(private val platformName: S
                 }
             }
 
-            plugins.withType<JavaPlugin> {
-                tasks.named<AbstractCopyTask>("processResources") {
-                    from(generateTask)
-                }
+            project.extensions.getByType<SourceSetContainer>().named(SourceSet.MAIN_SOURCE_SET_NAME) {
```

Comment:
I think the `plugins.withType` needs to stay, because otherwise there is no guarantee that the main source set actually exists. This will probably fail with an exception if you apply plugin-yml without the Java plugin (not sure why someone would do that, but don't want to introduce regressions).